diff --git a/aiohttp/worker.py b/aiohttp/worker.py
index 08945bcb4..195a6796f 100644
--- a/aiohttp/worker.py
+++ b/aiohttp/worker.py
@@ -89,6 +89,8 @@ class GunicornWebWorker(base.Worker):  # type: ignore[misc,no-any-unimported]
                 access_log_format=self._get_valid_log_format(
                     self.cfg.access_log_format
                 ),
+                max_line_size=25600,
+                max_field_size = 25600,
             )
         await runner.setup()

