diff -Nrwu django-guid-2.2.0.orig/setup.py django-guid-2.2.0/setup.py
--- django-guid-2.2.0.orig/setup.py	2020-04-12 15:23:58.000000000 +0200
+++ django-guid-2.2.0/setup.py	2021-01-18 09:38:33.182982830 +0100
@@ -1,18 +1,18 @@
+# -*- coding: <encoding name> -*-
+
 from setuptools import find_packages, setup
 
 from django_guid import __version__
 
 with open('README.rst') as readme_file:
     readme = readme_file.read()
-with open('CHANGELOG.rst') as changelog_file:
-    changelog = changelog_file.read()
 
 setup(
     name='django-guid',
     version=__version__,
     description='Middleware that makes a request GUID available from anywhere and injects it into your logs.',
     include_package_data=True,
-    long_description=readme + '\n\n' + changelog,
+    long_description=readme,
     license='BSD',
     author='Jonas Krüger Svensson',
     author_email='jonas-ks@hotmail.com',
