From 68369a6dbae71a9314ac0ecc8b88c435600cb4e9 Mon Sep 17 00:00:00 2001
From: Johan Andersson <johan.andersson@lime.tech>
Date: Tue, 3 Dec 2024 00:43:31 +0100
Subject: [PATCH] build: remove cargo.toml files from wheels

---
 pyproject.toml | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 4266e3bd5ba4..0378e78815e3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -101,15 +101,17 @@ include = [
     "src/_cffi_src/**/*.c",
     "src/_cffi_src/**/*.h",
 
-    "**/Cargo.toml",
-    "**/Cargo.lock",
+    "Cargo.toml",
+    "Cargo.lock",
+    "src/rust/**/Cargo.toml",
+    "src/rust/**/Cargo.lock",
     "src/rust/**/*.rs",
 
     "tests/**/*.py",
 ]
 exclude = [
     "vectors/**/*",
-    "src/rust/target/**/*",
+    "target/**/*",
     "docs/_build/**/*",
     ".github/**/*",
     ".readthedocs.yml",
