From baa901eef086a4b781924dfb0aa0e3c14ea0231e Mon Sep 17 00:00:00 2001
From: Matthias Dellweg <mdellweg@redhat.com>
Date: Wed, 9 Sep 2020 13:39:05 +0200
Subject: [PATCH] FIPS: Mark use of MD5 not security relevant

---
 django/db/backends/utils.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/django/db/backends/utils.py b/django/db/backends/utils.py
index 597e6b3a9d..0d0b16b136 100644
--- a/django/db/backends/utils.py
+++ b/django/db/backends/utils.py
@@ -213,12 +213,22 @@ def truncate_name(identifier, length=None, hash_len=4):
     return '%s%s%s' % ('%s"."' % namespace if namespace else '', name[:length - hash_len], digest)
 
 
+try:
+    hashlib.md5()
+except ValueError:
+    def _insecure_md5():
+        return hashlib.md5(usedforsecurity=False)
+else:
+    def _insecure_md5():
+        return hashlib.md5()
+
+
 def names_digest(*args, length):
     """
     Generate a 32-bit digest of a set of arguments that can be used to shorten
     identifying names.
     """
-    h = hashlib.md5()
+    h = _insecure_md5()
     for arg in args:
         h.update(arg.encode())
     return h.hexdigest()[:length]
-- 
2.28.0

