From 16d69707494bc9a0c70e49d9093c0b1b62deb58a Mon Sep 17 00:00:00 2001
From: Brian Bouterse <bmbouter@redhat.com>
Date: Wed, 3 Mar 2021 15:35:22 -0500
Subject: [PATCH] Adjusts worker timeout to 300 seconds

---
 pulpcore/tasking/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pulpcore/tasking/constants.py b/pulpcore/tasking/constants.py
index 4536c6816..f3c4eeb1c 100644
--- a/pulpcore/tasking/constants.py
+++ b/pulpcore/tasking/constants.py
@@ -4,7 +4,7 @@ TASKING_CONSTANTS = SimpleNamespace(
     # The name of resource manager entries in the workers table
     RESOURCE_MANAGER_WORKER_NAME="resource-manager",
     # The amount of time (in seconds) after which a worker process is considered missing.
-    WORKER_TTL=30,
+    WORKER_TTL=300,
     # The amount of time (in seconds) between checks
     JOB_MONITORING_INTERVAL=5,
 )
-- 
2.26.2

