commit ca540e8c70ee6238da0e928bdc6d6b5df3f34e52 (HEAD -> 3.11)
Author: David Davis <daviddavis@redhat.com>
Date:   Tue Apr 20 15:33:46 2021 -0400

    Handle tasking race condition when cleaning up reservations
    
    fixes #8603

diff --git a/CHANGES/8603.bugfix b/CHANGES/8603.bugfix
new file mode 100644
index 000000000..65b483a49
--- /dev/null
+++ b/CHANGES/8603.bugfix
@@ -0,0 +1 @@
+Handled a tasking race condition where cleaning up resource reservations sometimes raised an IntegrityError.
diff --git a/pulpcore/app/models/task.py b/pulpcore/app/models/task.py
index 678cb64bc..2647b0c91 100644
--- a/pulpcore/app/models/task.py
+++ b/pulpcore/app/models/task.py
@@ -8,6 +8,7 @@ from gettext import gettext as _
 
 from django.contrib.postgres.fields import JSONField
 from django.db import models, transaction
+from django.db.utils import IntegrityError
 from django.utils import timezone
 from rq.job import get_current_job
 
@@ -427,7 +428,11 @@ class Task(BaseModel, AutoDeleteObjPermsMixin, AutoAddObjPermsMixin):
         for reservation in self.reserved_resources.all():
             TaskReservedResource.objects.filter(task=self.pk).delete()
             if not reservation.tasks.exists():
-                reservation.delete()
+                try:
+                    reservation.delete()
+                except IntegrityError:
+                    # other tasks have added reservations for this resource
+                    pass
 
 
 class TaskGroup(BaseModel):
