--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2 +2 @@
-requires = ["hatchling", "hatch-fancy-pypi-readme", "uv-dynamic-versioning>=0.14.0"]
+requires = ["hatchling", "hatch-fancy-pypi-readme"]
@@ -5,9 +4,0 @@
-[tool.hatch.version]
-source = "uv-dynamic-versioning"
-
-[tool.uv-dynamic-versioning]
-vcs = "git"
-style = "pep440"
-bump = true
-fallback-version = "0.0.0"
-
@@ -15,0 +7 @@
+version = "2.12.0"
@@ -44,3 +35,0 @@
-dynamic = ["readme", "version", "dependencies"]
-
-[tool.hatch.metadata.hooks.uv-dynamic-versioning]
@@ -49 +38 @@
-    "httpcore2=={{ version }}; sys_platform != 'emscripten'",
+    "httpcore2==2.12.*; sys_platform != 'emscripten'",
@@ -54,0 +44 @@
+dynamic = ["readme"]
